### PR TITLE
Fix FindCuDNN.cmake to look environment variable correctly

### PR DESCRIPTION
**Pull request by sonots**
1. Fix FindCuDNN.cmake to look environment variable correctly
2. cosmetic changes
3. I propose to add support of CUDNN_ROOT_DIR environment variable.

ref. https://cmake.org/cmake/help/v3.5/command/if.html for 1.

> if(<variable|string>)
> True if given a variable that is defined to a value that is not a false constant. False otherwise. (Note macro arguments are not variables.)

This means that `if($ENV{CUDNN_INCLUDE_DIR})` requires a variable defined by the value of env variable must exist. But, a variable whose name is like a directory path does not exist.

Use `if(DEFINED <variable>)` instead.

Please let me know if 3. is not acceptable. I will revert it.

### CODE REVIEW

File: cmake/Modules/FindCuDNN.cmake [hunk 1]
```diff
@@ -13,30 +13,34 @@
...
 
-set(CUDNN_ROOT_DIR "" CACHE PATH "Folder contains NVIDIA cuDNN")
+if(DEFINED ENV{CUDNN_ROOT_DIR})
+  set(CUDNN_ROOT_DIR $ENV{CUDNN_ROOT_DIR})
+else()
+  set(CUDNN_ROOT_DIR "" CACHE PATH "Folder contains NVIDIA cuDNN")
+endif()
 
-if($ENV{CUDNN_INCLUDE_DIR})
+if(DEFINED ENV{CUDNN_INCLUDE_DIR})
```
Comment (zou3519) [inline cmake/Modules/FindCuDNN.cmake]: Is the DEFINED necessary?

Comment (sonots) [reply cmake/Modules/FindCuDNN.cmake]: Yes, it is the fix.

**Review by soumith — APPROVED** [review]

**Comment by soumith** [comment]
findcudnn.cmake was removed from master.

The logic for CuDNN detection now lives here (and it has the DEFINED fix): https://github.com/pytorch/pytorch/blob/master/cmake/public/cuda.cmake#L63-L101

**Comment by sonots** [comment]
Okay, thanks

**Pull request closed without merging**